astropy/utils/masked/function_helpers.py

```diff
@@ -99,6 +99,17 @@
 """Set of supported numpy functions with a 'like' keyword argument that dispatch
 on it (NEP 35).
 """
+if not NUMPY_LT_2_2:
+    # in numpy 2.2 these are auto detected by numpy itself
+    # xref https://github.com/numpy/numpy/issues/27451
+    SUPPORTED_NEP35_FUNCTIONS |= set()
+    UNSUPPORTED_FUNCTIONS |= {
```

Comment:
I'm confused - why do all these now need to be added to `UNSUPPORTED_FUNCTIONS`? Functions like `np.tri` are in fact supported.

p.s.  If this is just a mistake, maybe this suggests we need a test that ensures that the various sets/dicts have no intersection?

Reply:
I can't find a test or any explicit declaration that `np.tri` is indeed supported. What am I missing ?

Reply:
It's here: https://github.com/astropy/astropy/blob/f618f28da47975072bb4faa47df4d4357eae29a1/astropy/units/quantity_helper/function_helpers.py#L642-L649

Reply:
See also `TestArrayCreation`

Reply:
Oops, I was looking at `Quantity` - never even realized we were talking about `masked` here...  Sorry!!